src/Controller.ts

```diff
@@ -53,12 +53,25 @@ export const Controller = {
 		return engines;
 	},
 
-	getFilteredEngines: async (filteredNames: string[]): Promise<RuleEngine[]> => {
+	getUserFacingEngines: async (): Promise<RuleEngine[]> => {
 		const allEngines: RuleEngine[] = await Controller.getAllEngines();
-		const engines = allEngines.filter(e => filteredNames.includes(e.getName()));
+		const engines: RuleEngine[] = [];
+
+		for (const engine of allEngines) {
+			if (!engine.isCustomConfigBased()) {
+				engines.push(engine);
+			}
+		}
+
+		return engines;
+	},
+
+	getFilteredEngines: async (filteredNames: string[]): Promise<RuleEngine[]> => {
+		const userFacingEngines: RuleEngine[] = await Controller.getUserFacingEngines();
```

Comment:
We want to make sure we allow filtering only from user-facing engines and not from custom engines.

Reply:
I believe you, but I don't quite understand why this is true. Could you maybe add a comment or two explaining?

Reply:
"UserFacing" doesn't describe it well enough.
This is my take:
We shouldn't be listing `eslint-custom` and `pmd-custom` as an allowed engine filter value. Especially because the filter value won't make any sense without the `--eslintconfig` or `--pmdconfig` flags. And then we'll have to follow up with handling many different scenarios (for example, what if they filter on `eslint-custom` but provide `--pmdconfig` parameter value?)
Not allowing users to filter on the custom engines seems to help us avoid all the awkward scenarios.
However, with this implementation, there's still one weird scenario - if the user provides `--pmdconfig` and a `--engine pmd` filter, no rules get run.

Let me know what you think.